File: java/src/main/java/com/fengtian/qianer/atcoder/abc/abc162/ABC162_C.java
package com.fengtian.qianer.atcoder.abc.abc162;

import java.util.Scanner;

public class ABC162_C {

    public static void main(String[] args) {
        final Scanner sc = new Scanner(System.in);

        final int x = sc.nextInt();

        long ans = 0;
        for (int i = 1; i <= x; i++) {
            for (int j = 1; j <= x; j++) {
                for (int k = 1; k <= x; k++) {
                    ans += gcd(gcd(i, j), k);
                }
            }
        }

        System.out.println(ans);
    }

    private static long gcd(long a, long b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }
}

File: kotlin/src/abc/abc162/ABC162_C.kt
package abc.abc162

import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)

    val k = sc.nextInt()

    val ans = (1..k).map { a -> (1..k).map { b -> (1..k).map { c -> gcd(a, gcd(b, c)) }.sum() }.sum() }.sum()
    println(ans)
}

fun gcd(a: Int, b: Int) : Int {
    return if (b == 0)  a else gcd(b, a % b)
}
